File: app/src/main/java/com/akari/ppx/xp/hook/code/purity/ShareHook.java
package com.akari.ppx.xp.hook.code.purity;

import android.content.Context;

import com.akari.ppx.common.utils.XSP;
import com.akari.ppx.xp.hook.code.SuperbHook;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;

import static com.akari.ppx.common.constant.Prefs.SIMPLIFY_SHARE;

public class ShareHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		if (!XSP.get(SIMPLIFY_SHARE)) return;
		hookConstructor("com.sup.android.m_sharecontroller.ui.c", Context.class, "[Lcom.sup.android.i_sharecontroller.model.ShareInfo;",
				"[Lcom.sup.android.i_sharecontroller.model.OptionAction$OptionActionType;", "com.sup.android.i_sharecontroller.model.OptionAction$OptionActionListener", "com.sup.android.mi.feed.repo.bean.cell.AbsFeedCell", new XC_MethodHook() {
					@Override
					protected void beforeHookedMethod(MethodHookParam param) {
						param.args[1] = null;
					}
				});
		hookMethod("com.sup.android.m_sharecontroller.service.BaseShareService", "getAllShareletTypes", Context.class, int.class, XC_MethodReplacement.returnConstant(null));
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/purity/ShareHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.purity

import android.app.Activity
import android.content.Context
import com.akari.ppx.utils.*
import com.akari.ppx.xp.Init.absFeedCellClass
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.Init.shareViewClass
import com.akari.ppx.xp.hook.SwitchHook

class ShareHook : SwitchHook("simplify_share") {
    override fun onHook() {
        shareViewClass!!.hookBeforeConstructor(
            Context::class.java,
            "[Lcom.sup.android.i_sharecontroller.model.ShareInfo;",
            "[Lcom.sup.android.i_sharecontroller.model.OptionAction\$OptionActionType;",
            "com.sup.android.i_sharecontroller.model.OptionAction\$OptionActionListener",
            absFeedCellClass
        ) { param ->
            param.args[1] = null
        }
        "com.sup.android.m_sharecontroller.service.BaseShareService".replaceMethod(
            cl,
            "getAllShareletTypes",
            Context::class.java,
            Int::class.java
        ) { emptyList<Any>() }
        "com.sup.android.video.VideoDownLoadConfig".replaceMethod(
            cl,
            "getShowSuccessToast"
        ) { false }
        "com.sup.android.uikit.VideoDownloadProgressActivity".findClass(cl).apply {
            hookAfterMethod(
                declaredMethods.find { m ->
                    m.parameterTypes.size == 3 && m.parameterTypes[0] == Boolean::class.java
                            && m.parameterTypes[1] == Boolean::class.java && m.parameterTypes[2] == Boolean::class.java
                }?.name,
                Boolean::class.java,
                Boolean::class.java,
                Boolean::class.java
            ) { param ->
                param.args[1].check(true) {
                    showStickyToast(
                        "已保存至DCIM/${
                            "com.sup.android.business_utils.config.AppConfig".findClass(cl)
                                .callStaticMethodAs<String>("getDownloadDir")
                        }文件夹"
                    )
                    (param.thisObject as Activity).finish()
                }
            }
        }
    }
}